### PR TITLE
[JUJU-895] Work on cross compiled Juju builds.

### PR DESCRIPTION
**Pull request by tlm**
PR for work on cross compiled Juju builds in Jenkins.

### CODE REVIEW

File: jobs/common/pre-reqs.yaml [hunk 1]
```diff
@@ -26,20 +27,104 @@
...
+          properties-file: go-env.properties
+      - shell: |-
+          #!/bin/bash
+          set -ex
+          mkdir -p $GOPATH
+          rm -rf go-env.properties
+
+# set-common-environment is responsible for setting up a common environment for
+# our jobs. This shouldn't be used as a splat everything in builder that grows
+# out of control. It should be simple and only provide constructs that the vase
```
Comment (wallyworld) [inline jobs/common/pre-reqs.yaml:73]: vast

File: jobs/common/pre-reqs.yaml [hunk 2]
```diff
@@ -26,20 +27,104 @@
...
+# top of a builder list.
+- builder:
+    name: 'set-common-environment'
+    builders:
+      - shell: |-
+          #!/bin/bash
+          set -eux
+
+          touch build.properties
+          echo "BUILD_DIR=${WORKSPACE}/build" >> build.properties
```
Comment (wallyworld) [inline jobs/common/pre-reqs.yaml:86]: Seems like this is targetted at the jobs which compile stuff.
Is it really expected to be a common construct for other types of job?

File: jobs/common/pre-reqs.yaml [hunk 3]
```diff
@@ -26,20 +27,104 @@
...
+          echo "GOPATH=${WORKSPACE}/go-path" >> go-env.properties
+
+          cat go-env.properties
+      - inject:
+          properties-file: go-env.properties
+      - shell: |-
+          #!/bin/bash
+          set -ex
+          mkdir -p $GOPATH
+          rm -rf go-env.properties
```
Comment (wallyworld) [inline jobs/common/pre-reqs.yaml:69]: Can't this just be part of the previous shell build step? I guess you want to be able to run `rm go-env.properties`
We could use `properties-content` instead and not need an intermediate file. So have that as the first step and then the shell to create the dir. No properties file needed.

File: jobs/ci-run/ci-run-proving-grounds-tests.yml [hunk 4]
```diff
@@ -14,15 +14,22 @@
           fail: true
           type: absolute
     parameters:
-      - validating-string:
+      - string:
```
Comment (wallyworld) [inline jobs/ci-run/ci-run-proving-grounds-tests.yml:17]: This should stay as validating string, is there a reason it was changed?

File: jobs/ci-run/ci-run-master.yml [hunk 5]
```diff
@@ -123,12 +191,16 @@
...
+              SHORT_GIT_COMMIT=${SHORT_GIT_COMMIT}
+              GOVERSION=${GOVERSION}
           - project: "ci-proving-ground-tests"
             condition: SUCCESS
             current-parameters: true
             predefined-parameters: |-
               series=${series}
               BOOTSTRAP_SERIES=${series}
+              SHORT_GIT_COMMIT=${SHORT_GIT_COMMIT}
+              GOVERSION=${GOVERSION}
```
Comment (wallyworld) [inline jobs/ci-run/ci-run-master.yml:203]: delete me

File: jobs/ci-run/ci-run-master.yml [hunk 6]
```diff
@@ -123,12 +191,16 @@
...
```
Comment (wallyworld) [inline jobs/ci-run/ci-run-master.yml:195]: delete me

File: jobs/ci-run/ci-run-master.yml [hunk 7]
```diff
@@ -101,20 +101,88 @@
...
+          projects:
+            - name: publish-testing-streams-aws
               current-parameters: true
-            - name: simplestreams
+              predefined-parameters: |-
+                SHORT_GIT_COMMIT=${SHORT_GIT_COMMIT}
+            - name: publish-testing-streams-azure
               current-parameters: true
+              predefined-parameters: |-
+                SHORT_GIT_COMMIT=${SHORT_GIT_COMMIT}
```
Comment (wallyworld) [inline jobs/ci-run/ci-run-master.yml:184]: delete me

File: jobs/ci-run/ci-run-master.yml [hunk 8]
```diff
@@ -101,20 +101,88 @@
...
+      - multijob:
+          name: "Simple Streams Publish"
+          condition: SUCCESSFUL
+          execution-type: SEQUENTIALLY
+          projects:
+            - name: publish-testing-streams-aws
               current-parameters: true
-            - name: simplestreams
+              predefined-parameters: |-
+                SHORT_GIT_COMMIT=${SHORT_GIT_COMMIT}
```
Comment (wallyworld) [inline jobs/ci-run/ci-run-master.yml:180]: etc

File: jobs/ci-run/build/streams.yaml [hunk 9]
```diff
@@ -179,7 +108,7 @@
           # NOTE: this sync removes the "build-$SHA" indirection and effectively
           # pulls the binaries one folder up (under agent/)
           ${SCRIPTS_DIR}/azure_publish_tools.py sync \
-            ${WORKSPACE}/agent/build-${SHORT_GIT_COMMIT}/ \
+            ${WORKSPACE}/agent/build-${JUJU_VERSION}-${SHORT_GIT_COMMIT}/ \
```
Comment (wallyworld) [inline jobs/ci-run/build/streams.yaml:111]: Why do we need to add juju version? The git sha is all that's needed right?

File: jobs/ci-run/build/agent.yaml [hunk 10]
```diff
@@ -0,0 +1,68 @@
+## Jobs and builders for handling agent json
+- job:
+    name: "generate-agent-testing-streams"
```
Comment (wallyworld) [inline jobs/ci-run/build/agent.yaml:3]: not sure why this isn't in streams.yml?

File: jobs/common/pre-reqs.yaml (hunk 2 above)
Comment (tlm) [reply jobs/common/pre-reqs.yaml:86]: yeah all the integration tests use this directory.

File: jobs/ci-run/build/streams.yaml (hunk 9 above)
Comment (tlm) [reply jobs/ci-run/build/streams.yaml:180]: It is needed as it has the build number in it and re-running jobs with different build numbers will not always use the correct artifacts. Doing it this way ensures that the correct artifacts are used through out the whole run no matter what.

**Pull request merged**